src/tools/attack/attack.cpp

```diff
@@ -106,31 +125,57 @@ UdpUnreachCallback(
 {
 }
 
-void RunAttackRandom(CXPLAT_SOCKET* Binding, uint16_t Length, bool ValidQuic)
+_IRQL_requires_max_(DISPATCH_LEVEL)
+_Function_class_(CXPLAT_ROUTE_RESOLUTION_CALLBACK)
+void
+ResolveRouteComplete(
+    _Inout_ void* Context,
+    _When_(Succeeded == FALSE, _Reserved_)
+    _When_(Succeeded == TRUE, _In_reads_bytes_(6))
+        const uint8_t* PhysicalAddress,
+    _In_ uint8_t PathId,
+    _In_ BOOLEAN Succeeded
+    )
+{
+    CallbackContext* CContext = (CallbackContext*) Context;
+    CxPlatResolveRouteComplete(nullptr, CContext->Route, PhysicalAddress, 0);
+    CxPlatEventSet(CContext->Event);
+}
+
+void RunAttackRandom(CXPLAT_SOCKET* Binding, uint16_t Length, bool ValidQuic, bool TCP = false)
 {
     CXPLAT_ROUTE Route = {0};
     CxPlatSocketGetLocalAddress(Binding, &Route.LocalAddress);
-    Route.RemoteAddress = ServerAddress;
+    CxPlatSocketGetRemoteAddress(Binding, &Route.RemoteAddress);
+    CallbackContext Context = {&Route, };
+    QUIC_STATUS Status = CxPlatResolveRoute(Binding, &Route, 0, &Context, ResolveRouteComplete);
+    if (Status == QUIC_STATUS_PENDING) {
+        CxPlatEventInitialize(Context.Event, FALSE, FALSE);
+        if (!CxPlatEventWaitWithTimeout(Context.Event, (DWORD)TimeoutMs)) {
```

Comment:
You're not cleaning up the event if you timeout.

Reply:
Fixed.